### DISCUSSION TITLE
onAuthStateChanged event not firing in webpack scenario

### DISCUSSION THREAD

**Discussion by ProjectMoon** [supabase/supabase#1062] [Questions]
I've got a multi page static website that I'm experimenting with conversion from firebase. It's very simple, and basically static info with some extra information displayed if the user is logged in.

I have one client instance created in a typescript file, which is imported elsewhere. I've registered 2 event handlers for auth state changes in separate files. The first is in a file imported by the second file. In this scenario, only the first handler fires. But I inspected the auth object and both handlers are there. My only guess is that handler is registered, event fires, then other handler registered.

Is there some way I can defer client initialization or something until I've registered all my handlers?

**Comment by kiwicopple** [comment]
Hey @ProjectMoon - could you please share a repository with the code that you're describing? It will be a bit easier to debug if we have a reproducible environment

**Reply by ProjectMoon** [reply]
Yeah, I will have to craft an example. It's all entwined together at the moment. Will try to make a small self contained example. In the meantime, how does the event handler work? Does it buffer sign in events? Or does it try to sign in when you create the client?

**Comment by ProjectMoon** [comment]
My current workaround for this is to just initialize the supabase client cleanly in every page script, rather than importing an already-initialized one from another import. This might give me 2 or 3 client instances on a page, but does not cause the issue described in the OP.
